tests/Behat/Gherkin/Cucumber/CompatibilityTest.php

```diff
@@ -82,11 +83,15 @@ public function testFeaturesParseTheSameAsCucumber(\SplFileInfo $file)
 
     public static function cucumberFeatures()
     {
-        foreach (new \FilesystemIterator(self::TESTDATA_PATH . '/good') as $file) {
-           if ($file->isFile() && $file->getExtension() === 'feature') {
-                yield $file->getFilename() => [$file];
+        $files = array();
```

Comment:
this is not an array of files but an array of test cases

Reply:
also, yield is perfectly fine on PHP 5.6 so you could revert this

Reply:
The lowest version of phpunit we support doesn't allow a generator here

Reply:
that's only a bug in 5.7.0, which was fixed in 5.7.1. Bump the min version to 5.7.1 for the 5.7 branch (or better to the latest 5.7.x patch release) and you're OK.